### ISSUE TITLE
Use detr for one class

### ISSUE THREAD

**Issue by nedarov** [facebookresearch/detr#249]
I have a dataset of a specific class outside COCO classes and want to train detr for that object detector. Also, I want to use COCO dataset as samples (negative samples). I want to create my annotation json files using these two datasets.
For my own dataset samples, I will set category_id to 0. However, my question is for coco samples, I don't want to have any bbox, segmentation and category_id for these samples. How should I proceed? I can set bbox to an empty list. But wanted to make sure it works. I see in another thread: 

https://github.com/facebookresearch/detr/issues/82#issuecomment-643164035

boxes = torch.empty((0, 4), dtype=torch.float32)
labels = torch.empty((0,), dtype=torch.int64)

But I don't want to do it in my dataset class and want to write it in JSON file and Object of type Tensor is not JSON serializable.

**Comment by ruodingt** [comment]
I didn't see much point including COCO images into the training without using the bbox annotations.

All the `no_object` / `background` is negative examples for you in your own dataset.

**Comment by alcinos** [comment]
I'm not exactly sure what you want to do, but DETR supports empty annotations for a given image (there are some of those in the coco dataset).
Follow some guide (eg: https://www.immersivelimit.com/tutorials/create-coco-annotations-from-scratch/#coco-dataset-format) to create your dataset in the coco format (you'll simply have no annotations with the image_id of the image that should be empty)

I believe I have answered your question and as such I'm closing this. If you have further questions, feel free to reach out.

**Comment by darsh10** [comment]
What are all the different losses which are logged out? Even with JUST ONE CLASS, there are several loss_0, loss_1,...,loss_3 printed out. What is all that?

**Comment by ran337287** [comment]
I wonder if the negative samples are mainly obtained by images without objects.

boxes = torch.empty((0, 4), dtype=torch.float32)
labels = torch.empty((0,), dtype=torch.int64)
So this setting does not seem to be calculated by the loss function?

**Comment by ran337287** [comment]
I wonder if the negative samples are mainly obtained by images without targets. 

boxes = torch.empty((0, 4), dtype=torch.float32)
labels = torch.empty((0,), dtype=torch.int64)
So this setting does not seem to be calculated by the loss function?
